### ISSUE TITLE
Validation fails with failFast=true for Repeated field with 2 rules

### ISSUE THREAD

**Issue by npapadiochos** [telus-oss/protovalidate-net#21]
Hello,

In a proto3 file, I have the following message:

```
message QueryTypeRequestBody{
  repeated SystemCriteria  systemCriteria = 1 [(buf.validate.field) = {repeated:{min_items:1, max_items:2}}];
}
```

I then construct a request using 3 systemCriteria fields, and I feed it into the validator with failFast = true:

```
var queryValidationResult = protoValidator.Validate(queryRequest, failFast: true);
```

This should be triggered by the validator as a violation, but it doesn't (queryValidationResult.Violations.Count = 0).

It seems to me that this is a bug located at:
src/ProtoValidate/Internal/Cel/CompiledProgramsEvaluator.cs

![Image](https://github.com/user-attachments/assets/74193711-313c-47a2-8cb6-e9b6639694d3)

Here, we break the foreach loop, taking only the failFast value into account.
I would expect to break out of the loop when there is also a violation - something like:

```
...
if (failFast && violationList.Any())
{
    break;
}
...
```

In the first loop iteration, the min_items:1 validation rule is passed, since there are 3 systemCriteria fields in my request.
But then, without any violation count check, we exit prematurely, and there is no room for the max_items:2 rule to be checked by the validator.

What do you think about this?
I am expecting your comments - Thank you.

**Comment by LamarLugli** [comment]
It is definitely a bug.  I have a fix for it that I'll publish soon.

**Comment by LamarLugli** [comment]
This bug is fixed in release 0.3.3.

**Comment by npapadiochos** [comment]
I confirm that It works now as expected - Thank you!
